index.html

```diff
@@ -1939,29 +2030,32 @@
             <a href="#steps-pushMessage"><code>pushMessage()</code> algorithm</a>,
             or no active timers associated with it, then terminate this
             algorithm.
-            <p class="note">
-              This step also covers the case when the timer has already fired,
-              which also means that the message buffer is invalid, and is to be
-              replaced by the next invocation of <code>pushMessage()</code>.
-            </p>
           </li>
           <li>
-            Otherwise, if there is an active timer associated with a message set
-            to be pushed during the next NFC communication, then execute the
+            Otherwise, if there is an active timer <var>timer</var>
+            <a href="#adapter-timer">associated</a> with the current
+            <a>NFC adapter</a> for a message set to be pushed, then execute the
             following sub-steps:
-            <ul>
+            <ol>
               <li>
-                Remove the message that has been set for pushing.
+                Reset <var>timer</var>.
               </li>
               <li>
-                Cancel the timer associated with <code>messagePush()</code>.
+                Set the <var>tagMessage</var> variable
+                <a href="#adapter-tagmessage">associated</a> with the current
+                <a>NFC adapter</a> to <code>null</code>.
               </li>
-            </ul>
-          </li>
-          <li>
-            Reject the pending <var>promise</var> in the
-            <a href="#steps-pushMessage"><code>pushMessage()</code> algorithm</a>
-            with <code>"<a>AbortError</a>"</code>.
+              <li>
+                Set the <var>peerMessage</var> variable
+                <a href="#adapter-peermessage">associated</a> with the current
+                <a>NFC adapter</a> to <code>null</code>.
+              </li>
+              <li>
+                Reject the pending <var>promise</var> in the
+                <a href="#steps-pushMessage"><code>pushMessage()</code></a>
```

Comment:
This is a little awkward, but not bad enough that it absolutely needs to be fixed. The timer isn't intrinsically connected to any particular executing `pushMessage()` execution, so strictly it doesn't make sense to refer to "the" pending promise from the pushMessage() algorithm: there could be 2. But that algorithm also rejects its promise when `cancelPush()` is called, and it's clear enough what you mean here.